Makefile

```diff
@@ -1,28 +1,90 @@
+DEPS := "wget git go docker"
 
-all: build_local
+BINARY := autospotting
+BINARY_PKG := ./core
+COVER_PROFILE := /tmp/coverage.out
+BUCKET_NAME ?= cloudprowess
+LOCAL_PATH := build/s3/dv
 
-clean:
-	./build_helper.sh clean
+DOCKER_IMG := eawsy/aws-lambda-go
 
-prepare_bindata:
-	./build_helper.sh prepare_bindata
+GO_FILES := $(shell find ./ -name '*.go')
 
-build_lambda_binary: prepare_bindata
-	./build_helper.sh build_lambda_binary
+BINDATA_DIR := data
+BINDATA_FILE := generated_bindata.go
 
-prepare_upload_data: build_lambda_binary
-	./build_helper.sh prepare_upload_data
+SHA := $(shell git rev-parse HEAD | cut -c 1-7)
+BUILD := $(or $(TRAVIS_BUILD_NUMBER), $(TRAVIS_BUILD_NUMBER), $(SHA))
 
-# the following targets are only used for local development
+# upstream data
+EC2_INSTANCES_INFO_COMMIT_SHA := e655e36660bc617713c4ef9a1409cc65a209cb27
+INSTANCES_URL := "https://raw.githubusercontent.com/powdahound/ec2instances.info/$(EC2_INSTANCES_INFO_COMMIT_SHA)/www/instances.json"
 
-build_local: prepare_bindata
-	./build_helper.sh build_local
+all: build_local                                             ## Build the code
 
-upload: prepare_upload_data
-	./build_helper.sh upload
+clean:                                                       ## Remove installed packages/temporary files
+	go clean ./...
+	rm -rf $(BINDATA_DIR) $(BINDATA_FILE)
 
-test: build_local
-	./autospotting
+check_dep:                                                   ## Verify the system has all dependencies installed
+	@for DEP in $(shell echo "$(DEPS)"); do \
+		command -v "$$DEP" &>/dev/null \
+		|| (echo "Error: dependency '$$DEP' is absent" ; exit 1); \
+	done
+	@echo "all dependencies satisifed: $(DEPS)"
 
-cover:
-	./build_helper.sh calculate_coverage
\ No newline at end of file
+prepare_bindata: check_dep                                  ## Convert instance data into go file
```

Comment:
The "Convert instance data into go file" comment is misaligned with the rest, needs an extra whitespace before.

I would expect make to be smart enough to align them all.